### PR TITLE
Enhance FIM IT test_ambiguous_confs suite

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| #3591     |

## Description

This Issue aims to update the `test_fim/test_files/test_ambiguous_confs` IT suite to the new framework standard. 

During local executions for Windows agent, it was found some test failed because they were related to an unknown bug or had flaky behavior.  This tests have been `skipped` or `xfailed` and have been added to Issue #4077 so the bug can be tracked and can be fixed later.

### Updated

- Updated `test_fim/test_files/test_ambiguous_confs/test_ambiguous_whodata_thread.py`
- Updated `test_fim/test_files/test_ambiguous_confs/test_whodata_prevails_over_realtime.py`
- Updated `test_fim/test_files/test_ambiguous_confs/test_ignore_works_over_restrict.py`

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  | test_fim          | ⚫⚫⚫ | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11296041/3591-Manager.zip)  | Manager        | https://github.com/wazuh/wazuh-qa/pull/4004/commits/a99d78f98d977fd12699cd38238fc74232d0abc5         | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_fim          | ⚫⚫⚫ | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11296040/3591-Linux-Agent.zip) | Linux Agent        | https://github.com/wazuh/wazuh-qa/pull/4004/commits/8fb36f39fd5fc3b53b8b0ebf4e833bfb9330ef14        | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_fim          | ⚫⚫⚫ | ⚫⚫⚫ | Windows Agent         | https://github.com/wazuh/wazuh-qa/pull/4004/commits/a99d78f98d977fd12699cd38238fc74232d0abc5         | Nothing to highlight |
| @damarisg  (Reviewer)   |test_fim   |  [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/37708/) | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | Linux Agent          |  https://github.com/wazuh/wazuh-qa/pull/4004/commits/6e76e0ebb4283c348d90e206007ff2fd62c7f6c8       | Nothing to highlight |
| @damarisg  (Reviewer)   | test_fim   | [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/37707/) | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |Manager  | https://github.com/wazuh/wazuh-qa/pull/4004/commits/6e76e0ebb4283c348d90e206007ff2fd62c7f6c8        | Nothing to highlight |
| @damarisg  (Reviewer)   | test_fim  |  [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/37707/) | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | Windows Agent |  https://github.com/wazuh/wazuh-qa/pull/4004/commits/6e76e0ebb4283c348d90e206007ff2fd62c7f6c8       | Nothing to highlight |

### CODE REVIEW

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_fim/test_files/test_ambiguous_confs/test_ambiguous_whodata_thread.py [hunk 1]
```diff
@@ -149,14 +151,13 @@ def test_ambiguous_whodata_thread(whodata_enabled, tags_to_apply, get_configurat
...
-                                              '"File integrity monitoring real-time Whodata engine started" event')
+    if metadata['whodata_enabled']:
+        print("PRINT ENABLED")
+        detect_whodata_start(wazuh_log_monitor, timeout=T_10)
     else:
         with pytest.raises(TimeoutError):
-            wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
-                                    callback=callback_real_time_whodata_started)
+            detect_whodata_start(wazuh_log_monitor, timeout=T_10)
+            print("PRINT DIsABLED")
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/test_ambiguous_whodata_thread.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/test_ambiguous_whodata_thread.py [hunk 2]
```diff
@@ -149,14 +151,13 @@ def test_ambiguous_whodata_thread(whodata_enabled, tags_to_apply, get_configurat
...
     '''
-    check_apply_test(tags_to_apply, get_configuration['tags'])
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
 
-    if whodata_enabled:
-        wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_real_time_whodata_started,
-                                error_message='Did not receive expected '
-                                              '"File integrity monitoring real-time Whodata engine started" event')
+    if metadata['whodata_enabled']:
+        print("PRINT ENABLED")
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/test_ambiguous_whodata_thread.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    folder_index: 1
+    filename: regex_testfile
+    is_pattern: false
+    fim_mode: realtime
+
+- name: Ignore sregex over restrict regex - same regex (Whodata mode)
+  description: Check Whodata thread is disabled when last directory has it set to 'no'
+  configuration_parameters:
+    REALTIME: 'no'
+    WHODATA: 'yes'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    folder_index: 0
+    filename: regex_testfile
+    is_pattern: false
+    fim_mode: realtime
+
+- name: Ignore sregex over restrict regex - different regex (Whodata mode)
+  description: Check Whodata thread is disabled when last directory has it set to 'no'
+  configuration_parameters:
+    REALTIME: 'no'
+    WHODATA: 'yes'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_whodata_prevails_over_realtime.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+- name: Whodata thread started with realtime enabled
+  description: Check Whodata thread is enabled and is used with realtime enabled or disabled on same tag
+  configuration_parameters:
+    WHODATA: 'yes'
+    REALTIME: 'yes'
+  metadata:
+    fim_mode: whodata
+
+- name: Whodata thread started with realtime disabled
+  description: Check Whodata thread is enabled and is used with realtime enabled or disabled on same tag
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_whodata_prevails_over_realtime.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_whodata_prevails_over_realtime.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+- name: Whodata thread started with realtime enabled
+  description: Check Whodata thread is enabled and is used with realtime enabled or disabled on same tag
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_whodata_prevails_over_realtime.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

File: tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    folder_index: 0
+    filename: testfile
+    is_pattern: true
+    fim_mode: realtime
+
+- name: Ignore file pattern over restrict regex (Whodata mode)
+  description: Check Whodata thread is disabled when last directory has it set to 'no'
+  configuration_parameters:
+    REALTIME: 'no'
+    WHODATA: 'yes'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_ambiguous_confs/data/test_cases/cases_ignore_works_over_restrict.yaml]: Done in https://github.com/wazuh/wazuh-qa/pull/4004/commits/a5f0fdf882902e0866bdfefc9ce2ec871f5fa5fb

**Comment by Deblintrake09** [comment]
Closed. Branch recreated in #4121

**Pull request closed without merging**